rotkehlchen/chain/arbitrum_one/modules/umami/decoder.py

```diff
@@ -0,0 +1,236 @@
+            return 'MAIN'
+        elif event.balance.amount == fee_match_amount:
+            event.event_type = fee_event_type
+            event.event_subtype = HistoryEventSubType.FEE
+            event.notes = f'Umami execution cost of {event.balance.amount} ETH'
+            event.counterparty = CPT_UMAMI
+            return 'FEE'
+
+        return None
+
+    def _decode_deposit_request(self, context: DecoderContext, event: 'EvmEvent') -> REQUEST_EVENT_TYPE | None:  # noqa: E501
+        """Decode deposit request events.
+        Returns REQUEST_EVENT_TYPE identifying the type of event found
+        to allow event reshuffling later, or None if no event is found.
+        """
+        resolved_asset = event.asset.resolve_to_crypto_asset()
+        match_amount = asset_normalized_value(
+            amount=int.from_bytes(context.tx_log.data[0:32]),
+            asset=resolved_asset,
+        )
+        return self._decode_deposit_withdraw_request(
+            event=event,
+            match_amount=match_amount,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            event_notes=f'Deposit {match_amount} {resolved_asset.symbol} into Umami',
+            fee_match_amount=asset_normalized_value(
+                amount=context.transaction.value,
+                asset=resolved_asset,
+            ),
+            fee_event_type=HistoryEventType.DEPOSIT,
+        )
+
+    def _decode_withdraw_request(self, context: DecoderContext, event: 'EvmEvent') -> REQUEST_EVENT_TYPE | None:  # noqa: E501
+        """Decode withdraw request events.
+        Returns REQUEST_EVENT_TYPE identifying the type of event found
+        to allow event reshuffling later, or None if no event is found.
+        """
+        resolved_asset = event.asset.resolve_to_crypto_asset()
+        match_amount = asset_normalized_value(
+            amount=int.from_bytes(context.tx_log.data[32:64]),
+            asset=resolved_asset,
+        )
+        return self._decode_deposit_withdraw_request(
+            event=event,
+            match_amount=match_amount,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.RETURN_WRAPPED,
+            event_notes=f'Return {match_amount} {resolved_asset.symbol} to Umami',
+            fee_match_amount=asset_normalized_value(
+                amount=context.transaction.value,
+                asset=resolved_asset,
+            ),
+            fee_event_type=HistoryEventType.WITHDRAWAL,
+        )
+
+    def _decode_umami_vault_events(self, context: DecoderContext) -> DecodingOutput:
+        """Decode umami vault deposit/withdraw events.
+
+        A deposit/withdraw consists of two transactions, a request tx and an execution tx.
+        The event for deposit execution is not available in decoded_events here,
+        and must be transformed later via an ActionItem.
+        """
+        main_event = fee_event = None
```

Comment:
Never do that in python. It works in some cases, but not for mutable objects. As if you modify one you end up modifying the other too.

This is how you modify multiple objects explicitly and safely

```suggestion
        main_event, fee_event = None, None
```